clean_project/index.py

```diff
@@ -31,17 +29,48 @@ def index(config_path, index_path):
                         index_words[word.lower()][source_file.name].append(i)
     else:
         raise ValueError('Empty configuration file')
-    with open(index_path, 'w') as index_file:
-        json.dump(index_words, index_file, sort_keys=True, indent=4)
     return index_words
 
 
+def build_index(index_words):
+    """
+    Insert the index in a specific database for store all data
+    :param index_words: Object which contains the built index
+    """
+    connection = None
+    try:
+        connection = psycopg2.connect(database='index_db', user='gianluca', password='password')
+        cursor = connection.cursor()
+        query_update = "UPDATE index SET occurrences=%s WHERE word=%s AND file=%s;"
+
+        query_insert = ("INSERT INTO index (word, file, occurrences)"
+                        "SELECT %s, %s, %s"
+                        "WHERE NOT EXISTS (SELECT 1 FROM index WHERE word=%s AND file=%s);")
+        for word in index_words:
+            for file_path in index_words[word]:
+                data_update = (index_words[word].get(file_path), word, file_path)
```

Comment:
i understand after talking IRL what these queries are for, although i don't know whether they're the best way to do it. you're doing 2 queries, one of which is a nested query. pretty sure we could do better by retrieving record if it exists. if it exists, update the array; if it doesn't, create the record. After this, replace the old row with the data you created this way. this way you're guaranteed to do exactly 2 *simple* queries that will always have an effect.